src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs

```diff
@@ -29,10 +29,7 @@ public DynamicQueryRunner(DocumentDatabase database) : base(database)
         public override async Task ExecuteStreamQuery(IndexQueryServerSide query, DocumentsOperationContext documentsContext, HttpResponse response, BlittableJsonTextWriter writer,
             OperationCancelToken token)
         {
-            var index = await MatchIndex(query, false, token.Token);
-
-            if (index == null)
-                IndexDoesNotExistException.ThrowFor("There was no auto index able to handle streaming query"); // TODO arek - pretty sure we want to change that behavior
+            var index = await MatchIndex(query, true, token.Token);
```

Comment:
We need to wait here a lot more for the indexing to complete, no?
In this case, we probably explicitly intend to do that.

Reply:
1 minute? 

I wonder if we should force a newly created auto index to stop the its first indexing batch before the timeout of a query which triggered it happens (let's say after `(query_timeout) / 2`). This way we'll return at least some stale results. Otherwise if the indexing batch is still in progress we return no results.

Reply:
Good idea on both

Reply:
And not just for streaming, for regular ones as well

Reply:
OK